### PR TITLE
py-tensorflow: remove patch file for 2.16-rocm-enhanced

### PR DESCRIPTION
**Pull request by afzpatel**
The changes to make tensorflow with rocm support compatible with spack have been pushed:
https://github.com/ROCm/tensorflow-upstream/commit/c467913bf4411ce2681391f37a9adf6031d23c2c

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-tensorflow/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/py-tensorflow/package.py:419]: Did you intend to also remove this patch?

**Review by tldahlgren — COMMENTED** [review]
Confirmed patch URLs exist.  Just have a quick question about one of the two affected patches.

File: var/spack/repos/builtin/packages/py-tensorflow/package.py
Comment (afzpatel) [reply var/spack/repos/builtin/packages/py-tensorflow/package.py:419]: Yes I intend to remove that patch as well. That change was pushed here:
https://github.com/ROCm/tensorflow-upstream/commit/f4f4e8698b90755b0b5ea2d9da1933b0b988b111

**Comment by afzpatel** [comment]
I had to add an additional patch as some of the paths were still assuming all the rocm components were in a single ROCM_PATH. I'll try to push the changes in the new patch to the rocm/tensorflow-upstream eventually.
I also removed version 2.14-rocm-enhanced since it didn't seem to be working and since we already have the newer 2.16-rocm-enhanced version.

File: var/spack/repos/builtin/packages/py-tensorflow/package.py
Comment (tldahlgren) [reply var/spack/repos/builtin/packages/py-tensorflow/package.py:419]: Should've been clearer, did you intend to *also* remove the patch file from the Spack repository?

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Please remove the unused patch file from the package repository.

File: var/spack/repos/builtin/packages/py-tensorflow/package.py
Comment (afzpatel) [reply var/spack/repos/builtin/packages/py-tensorflow/package.py:419]: Ya I did, thanks for catching that

**Comment by afzpatel** [comment]
@spackbot rerun pipeline

**Review by tldahlgren — DISMISSED** [review]
Thanks.

**Comment by tldahlgren** [comment]
LGTM. @adamjstewart @aweits Do either of you want to comment on this PR before it is merged?

File: var/spack/repos/builtin/packages/py-tensorflow/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tensorflow/package.py]:
```suggestion
import os
```

File: var/spack/repos/builtin/packages/py-tensorflow/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-tensorflow/package.py]: We don't normally remove versions without deprecating them first

**Review by tldahlgren — APPROVED** [review]
I will defer to @adamjstewart on this PR.

File: var/spack/repos/builtin/packages/py-tensorflow/package.py
Comment (afzpatel) [reply var/spack/repos/builtin/packages/py-tensorflow/package.py]: I added it back. It's probably worth keeping that version since it's still working with external rocm.

**Review by adamjstewart — APPROVED** [review]
If I'm understanding correctly, the reason that these patches are being removed is because they were merged into the AMD release branches. AKA, the existing versions were working correctly until the patches were merged, then all builds started failing, and this PR is required to fix them. This lack of stability is the main reason we avoid using branches in Spack. Would be great if we could use stable, checksummable release tarballs or commit hashes in the future. Would be even better if we didn't need to use a fork for TF+ROCm at all, but I think that's already in the works. I'll try to restrain my complaints because I think you've heard them plenty of times before, but it would be really nice to get TF+ROCm in CI so we can detect and fix breakages like this.

**Comment by afzpatel** [comment]
> If I'm understanding correctly, the reason that these patches are being removed is because they were merged into the AMD release branches. AKA, the existing versions were working correctly until the patches were merged, then all builds started failing, and this PR is required to fix them. This lack of stability is the main reason we avoid using branches in Spack. Would be great if we could use stable, checksummable release tarballs or commit hashes in the future. Would be even better if we didn't need to use a fork for TF+ROCm at all, but I think that's already in the works. I'll try to restrain my complaints because I think you've heard them plenty of times before, but it would be really nice to get TF+ROCm in CI so we can detect and fix breakages like this.

The changes were merged only into the develop branch of https://github.com/ROCm/tensorflow-upstream. The rocm-enhanced branches were unaffected and py-tensorflow+rocm was still building successfully, but I ran into some other problems when testing that required some additional changes. Also, I didn't like the idea of having patch files that are almost 400 lines long which is why I wanted to remove them once the changes were commited. I've brought up using release tarballs instead of branches to the folks that maintain the ROCm tf fork, so hopefully we'll be using tarballs for future releases.

Since TF+ROCM is working with spack built ROCm binaries now, I think we can try adding that to the CI as well.

**Pull request merged**
